src/tools/tests/golden.arcs

```diff
@@ -16,3 +16,10 @@ particle Gold
   alias: writes {val: Text}
 
   collection: reads [{num: Number}]
+
+  typeVarIn: reads ~a with {amt: Number}
+  typeVarOut: writes ~a
+
+  // This type var shares a schema with `collection`. Will two different entities still be created?
```

Comment:
Yes!  These should be different entities, as the entity for typeVarS needs to have a private constructor.